src/utils/smartweave/utils.go

```diff
@@ -0,0 +1,31 @@
+package smartweave
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/warp-contracts/syncer/src/utils/arweave"
+	"github.com/warp-contracts/syncer/src/utils/tool"
+)
+
+func ValidateInteraction(tx *arweave.Transaction) error {
+	if tx.Format < 2 {
+		return errors.New("interaction should not have a format smaller than 2")
+	}
+
+	tags := tx.GetTagMap(TagAppName, TagContractTxId, TagInput)
```

Comment:
this assumes there's only one tag of each type, there can be many

Reply:
For example there can be many App-Name tags

Reply:
Creating a map seems to be an overkill, one iteration through tags should be enough.

Reply:
fixed